### PR TITLE
[WIP] Add script to record Gravity responses that can be used as fixtures.

### PR DESCRIPTION
**Pull request by alloy**
See https://github.com/artsy/antigravity/blob/add-fixtures/fixtures/gravity/README.md

### CODE REVIEW

File: fixtures/gravity/index.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+async function authenticate(email: string, password: string): Promise<string> {
+  try {
+    const response = await request({
+      method: "POST",
+      uri: "https://api.artsy.net/oauth2/access_token",
+      qs: {
+        // Eigen OSS keys https://github.com/artsy/eigen/blob/master/Makefile
+        client_id: "e750db60ac506978fc70",
+        client_secret: "3a33d2085cbd1176153f99781bbce7c6",
```
Comment (mzikherman) [inline fixtures/gravity/index.js]: may want to hide/rotate this... (public repo)

Comment (mzikherman) [inline fixtures/gravity/index.js]: ah it's the OSS ones!

giphy 1

File: fixtures/gravity/index.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
+// @flow
+import "babel-polyfill"
+
+import { spawnSync } from "child_process"
+import { basename, dirname, extname, join } from "path"
+import { writeFile } from "fs"
+
+import request from "request-promise"
```
Comment (craigspaeth) [inline fixtures/gravity/index.js]: FWIW I've found [superagent](https://github.com/visionmedia/superagent) to have a nicer API and has promises built in. #superminor

File: fixtures/gravity/index.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+// request.debug = true
+
+async function authenticate(email: string, password: string): Promise<string> {
+  try {
+    const response = await request({
```
Comment (craigspaeth) [inline fixtures/gravity/index.js]: https://github.com/visionmedia/superagent

File: fixtures/gravity/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+This script records responses from Gravity in production and stores them on disk. It determines the path to query for by
+the filename on disk.
+
+For example, to record the data of an artist at the `https://api.artsy.net/api/v1/artist/andy-warhol` URL, you would
+perform the following steps:
+
+```sh
+$ cd fixtures
+$ touch 'gravity/api/v1/artist/andy-warhol.json'
+$ yarn run gravity you@example.com your-secret-password
```
Comment (craigspaeth) [inline fixtures/gravity/README.md]: Neat! Didn't know Yarn also ran scripts like npm.

**Comment by craigspaeth** [comment]
💯  Very cool! I've been wanting to do this record-from-Gravity-thing for a long time. So many shiny things in this PR too! Flow, Yarn, ES2017 👏

**Comment by craigspaeth** [comment]
This is awesome! Curious what inspired this PR? Excited to see some attention given to antigravity 👍

**Comment by craigspaeth** [comment]
Ack sorry I merged too early and missed the WIP. I'll put up a revert PR 😅

**Comment by alloy** [comment]
Hahaha, trigger happy, are we?

Giphy

**Comment by alloy** [comment]
> Curious what inspired this PR?

Like you, I too have been thinking about recording Gravity responses and use them for testing for a while now.

Recently I was reminded about the usefulness of this [when I had accidentally removed a MP field](https://github.com/artsy/metaphysics/pull/475) that wasn’t covered by a test, which I can imagine because the amount of setup required for a test that just checks if 1 field in a Gravity response correctly maps to a MP response feels kinda pedantic and tiresome.

Friday morning I read [a message from @gnilekaw](https://artsy.slack.com/archives/testing/p1479502872000174) that describes exactly how I had been thinking of setting this up in the past and it spurred me to start with the lowest hanging fruit first, which is the data recordings.

A *WIP* PR for MP will be up shortly.

**Comment by craigspaeth** [comment]
[Nock](https://github.com/node-nock/nock) might also be an interesting module to check out, especially it's part about [recording](https://github.com/node-nock/nock#recording).

File: fixtures/gravity/index.js (hunk 1 above)
Comment (alloy) [inline fixtures/gravity/index.js]: @mzikherman Still, though, I’ve been wondering if I should be creating new OSS ones for projects like these (and Emission’s example app). Do you have any thoughts on that?

File: fixtures/gravity/index.js (hunk 2 above)
Comment (alloy) [inline fixtures/gravity/index.js]: Ah cool, I’ll take a look at that when continuing work on the script, like e.g. record positron responses.

**Pull request merged**